# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/CommonSocketControl.cpp
# Commit: 632eefa7e529
# Full Hash: 632eefa7e529e806bdbecfb50e23f33e2dd86e37
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-17 21:50:08
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Add a callback when the handshake is done. r=necko-reviewers,kershaw,keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123601
# ==============================================================================

diff -r fe6b8f41ca83 -r 632eefa7e529 security/manager/ssl/CommonSocketControl.cpp
--- a/security/manager/ssl/CommonSocketControl.cpp	Fri Sep 17 13:19:00 2021 +0000
+++ b/security/manager/ssl/CommonSocketControl.cpp	Fri Sep 17 13:19:00 2021 +0000
@@ -15,6 +15,7 @@
 #include "ssl.h"
 #include "mozilla/net/SSLTokensCache.h"
 #include "nsICertOverrideService.h"
+#include "nsITlsHandshakeListener.h"
 
 using namespace mozilla;
 
@@ -356,4 +357,10 @@
 }
 
 NS_IMETHODIMP
+CommonSocketControl::SetHandshakeCallbackListener(
+    nsITlsHandshakeCallbackListener* callback) {
+  return NS_ERROR_NOT_IMPLEMENTED;
+}
+
+NS_IMETHODIMP
 CommonSocketControl::DisableEarlyData(void) { return NS_ERROR_NOT_IMPLEMENTED; }